### ISSUE TITLE
Add support for `Process` classes to `QueryBuilder.append`

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#2400]
If I have the `CalcJob` class `PwCalculation`, which has entry point `aiida.calculations:quantumespresso.pw`, calling `QueryBuilder.append(PwCalculation).all()` should tell the query builder to return all nodes that have a type string `node.process.calculation.calcjob.CalcJobNode.` and a `process_type` string `aiida.calculations:quantumespresso.pw`. This is essentially an `AND` statement of these two values. Note that since `append` supports adding tuples of classes, e.g. `QueryBuilder.append((PwCalculation, ArithmeticAddCalculation))`, this should be unwrapped into two sub filters that are `OR`'ed, i.e. in pseudo SQL
```
        WHERE (
            type = 'node.process.calculation.calcjob.CalcJobNode.' AND
            process_type = 'aiida.calculations:quantumespresso.pw'
        ) OR (
             type = 'node.process.calculation.calcjob.CalcJobNode.' AND
             process_type = 'aiida.calculations:arithmetic.add'
        )
```
technically since the `process_type` should also support querying for sub classes, the filter on process type should be an `OR` of `process_type = 'aiida.calculations:quantumespresso.pw'` and `process_type like 'aiida.calculations:quantumespresso.pw.'`

**Comment by ltalirz** [comment]
Just to summarize the logic here:


 * parameter `cls`: need to provide a *class* (instances not allowed, as before)
   * if an instance is provided (instead of a class), raise an exception
   * if Process class is provided, get corresponding Node class and filter by `process_type`
   * if Node class is provided, try to infer `process_type`. If it cannot be found (e.g. not a ProcessNode), just use the `type` for filtering.
   * if a tuple of classes is provided, use combination logic mentioned in the issue description

 * parameter `type`: we would need to add an additional parameter `process_type`. At this point, the question arises whether we really need to have both `type` and `process_type` as parameters. Should they just be replaced by filters instead? @lekah 

 * parameter `subclassing`: if `True`, include subclasses both of `type` and `process_type`.

**Comment by sphuber** [comment]
Fixed in PR #2421
